R/runEmployment.R

```diff
@@ -0,0 +1,304 @@
+   # report_tech <- reportTechnology(gdx)
+    #report_tech <- collapseNames(report_tech)
+    
+    # variables needed - capital and fixed costs of technologies
+    var <- c("Tech|Electricity|Coal|Pulverised Coal w/o CC|Capital Costs",
+             "Tech|Electricity|Gas|Combined Cycle w/o CC|Capital Costs",
+             "Tech|Electricity|Hydro|Capital Costs",
+             "Tech|Electricity|Nuclear|Capital Costs",
+             "Tech|Electricity|Solar|PV|Capital Costs",
+             "Tech|Electricity|Solar|CSP|Capital Costs",
+             "Tech|Electricity|Wind|Onshore|Capital Costs",
+             "Tech|Electricity|Wind|Offshore|Capital Costs",
+             "Tech|Electricity|Biomass|Combined Heat and Power w/o CC|Capital Costs",
+             "Tech|Electricity|Storage|Battery|For PV|Capital Costs",
+             "Tech|Electricity|Oil|DOT|Capital Costs",
+             "Tech|Electricity|Geothermal|Capital Costs",
+             
+             "Tech|Electricity|Coal|Pulverised Coal w/o CC|OM Cost|fixed",
+             "Tech|Electricity|Gas|Combined Cycle w/o CC|OM Cost|fixed",
+             "Tech|Electricity|Hydro|OM Cost|fixed",
+             "Tech|Electricity|Nuclear|OM Cost|fixed",
+             "Tech|Electricity|Solar|PV|OM Cost|fixed",
+             "Tech|Electricity|Solar|CSP|OM Cost|fixed",
+             "Tech|Electricity|Wind|Onshore|OM Cost|fixed",
+             "Tech|Electricity|Wind|Offshore|OM Cost|fixed",
+             "Tech|Electricity|Biomass|Combined Heat and Power w/o CC|OM Cost|fixed",
+             "Tech|Electricity|Oil|DOT|OM Cost|fixed",
+             "Tech|Electricity|Geothermal|OM Cost|fixed",
+             "Tech|Electricity|Storage|Battery|For PV|OM Cost|fixed", 
+             NULL
+    )
+    # only capital costs
+    report_tech <- input_mif_mp[,,var]
+    cap_costs <- report_tech[getRegions(x), seq(2015, 2050, 5), "Capital Costs",pmatch=TRUE]# selecting years until 2050 and removing "World" region
+
+    cap_costs <- cap_costs[, , ] / setYears(cap_costs[, "y2020", ], NULL) # costs relative to 2020
+    
+    # Evolution of employment factor (CI & Manf) with time depends on capital costs
+    # the job intensity decreases with capital costs for techs, relative to 2015
+    var_in_x <- getItems(x, dim = 3.1) # variables from employment factor magpie object
+    var_in_rem <- c("Coal", "Gas", "Nuclear", "Biomass", "Hydro", "Hydro", "Wind|Onshore", "Wind|Offshore", "Solar|PV",
+                    "Geothermal", "Solar|CSP", "Oil", "Solar|PV", "Storage") # variables from remind (order matters)
+    var_comb <- paste(var_in_x, var_in_rem, sep = ".") # concatenating the two for looping to work
+    
+    for (i in var_comb) {
+      x[getRegions(x), , sub("\\..*", "", i)][, , c("CI", "Manf")] <- x[getRegions(x), , sub("\\..*", "", i)][, , c("CI", "Manf")] * setNames(cap_costs[getRegions(x), getYears(x), sub(".*\\.", "", i), pmatch = TRUE], NULL)
```

Comment:
Please try to split up lines that are longer than 120 characters into smaller simpler steps to improve readability. If that is not possible please add line breaks.

Reply:
Did where ever possible without affecting code readability